src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -464,171 +468,240 @@ subroutine init_atm_static(mesh, dims, configs)
+         do j = 1 + tile_bdr, tile_ny + tile_bdr, 1
+             do k = 1 + tile_bdr, tile_nx + tile_bdr, 1
 
-                ncat(int(rarray(i,j,1)),iPoint) = ncat(int(rarray(i,j,1)),iPoint) + 1
+                !
+                ! The MODIS dataset has zeros at the north pole, and possibly other places, so
+                ! skip any value that is below the category_min or category_max as reported in
+                ! the index file.
+                !
+                if (tile % tile(k,j,1) <= category_min .or. tile % tile(k,j,1) >= category_max) then
+                    cycle
+                end if
 
-             end if
-          end if
-       end do
-       end do
+                call mgr % tile_to_latlon(tile, j, k, lat_pt, lon_pt)
+                call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat_pt, lon_pt)
+                call mpas_kd_search(tree, (/xPixel, yPixel, zPixel/), res)
 
-    end do
+                !
+                ! For all but the outermost boundary cells, we can safely assume that the nearest
+                ! model grid cell contains the pixel (else, a different cell would be nearest)
+                !
+                if (bdyMaskCell(res % id) < nBdyLayers) then
+
+                   !write(0,*) tile % tile(k,j,1)
+                   ncat(int(tile % tile(k,j,1)), res % id) = ncat(int(tile % tile(k,j,1)), res % id) + 1
+
+                   if (res % id <= nCellsSolve) then
+                       all_pixels_mapped_to_halo_cells = .false.
+                   end if
+
+                ! For outermost boundary cells, additional work is needed to verify that the pixel
+                ! actually lies within the nearest cell
+                else
+                   if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                    nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                      ncat(int(tile % tile(k,j,1)), res % id) = ncat(int(tile % tile(k,j,i)), res % id) + 1
+
+                      if (res % id <= nCellsSolve) then
+                          all_pixels_mapped_to_halo_cells = .false.
+                      end if
+                   end if
+                end if
+             end do
+         end do
+
+         tile % is_processed = .true.
+
+         if (.not. all_pixels_mapped_to_halo_cells) then
+             ierr = mgr % push_neighbors(tile)
+             if (ierr /= 0) then
+                 call mpas_log_write("Error pushing the tile neighbors of: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+             endif
+         end if
+
+     end do
  end do
 
- do iCell = 1,nCells
-    lu_index(iCell) = 1
-    do i = 2,ismax_lu
-       if(ncat(i,iCell) > ncat(lu_index(iCell),iCell)) then
-          lu_index(iCell) = i
-       end if
-    end do
+ do iCell = 1, nCells
+     lu_index(iCell) = 1
+     do i = 2, ismax_lu
```

Comment:
@mgduda was there a reason for looping between 2 and `ismax_lu` here instead of 1? Both of the landuse datasets and soiltop dataset have `category_min` set to 1 in their index files.

Reply:
If we looped from 1 to `ismax_lu`, the first iteration would check whether `ncat(1,iCell) > ncat(1,iCell)`, which is always false and therefore unnecessary.

Actually, though, we could probably use `maxloc` here rather than a loop to find the index of the maximum.